src/exchanges/cache.ts

```diff
@@ -152,13 +153,13 @@ const afterQuery = (
   resultCache: ResultCache,
   operationCache: OperationCache
 ) => (response: OperationResult) => {
-  const { operation, data } = response;
+  const { operation, data, error } = response;
 
   if (data === undefined) {
     return;
   }
 
-  resultCache.set(operation.key, response);
+  resultCache.set(operation.key, { operation, data, error });
```

Comment:
Hm, this is probably a change we need for another PR, but it isn't safe to store the `error` I think. On the other hand it's obviously also not safe to store the data at all when there's an `error`. I suppose we have to revisit this

Reply:
yes, sorry, you're right. I'm just noticing somthing completely unrelated 😅  we may not want to cache the result when there's been a more severe error. For instance when the `CombinedError` has a `networkError`, but that's definitely sth for another PR. This change LGTM :+1: